### ISSUE TITLE
Don't know how to build task 'assets:clean_expired'

### ISSUE THREAD

**Issue by runemadsen** [ndbroadbent/turbo-sprockets-rails3#88]
Suddenly I'm getting this error when pushing to Heroku. This happened after bumping to gem version `gem 'turbo-sprockets-rails3', '~> 0.3.11'`. Using Rails version `3.2.13`.

Any ideas of what could cause this?

```
Copied binary asset to glyphicons.png
Copied binary asset to glyphicons.svg
Copied binary asset to logo-white.svg
Copied binary asset to logo.svg
Generated non-digest assets in 623ms
Asset precompilation completed (457.15s)
Running: rake assets:clean_expired
rake aborted!
Don't know how to build task 'assets:clean_expired'
 (See full trace by running task with --trace)
-----> Discovering process types
Procfile declares types -> web
Default types for Ruby  -> console, rake, worker
```

**Comment by pboling** [comment]
Using Rails 3.2.18 with 

`gem 'turbo-sprockets-rails3',   '~> 0.3.11'`

I am having the same error.

**Comment by nfriend21** [comment]
+1

**Comment by gaugau** [comment]
+1

`RAILS_GROUP=assets rake -T | grep asset` will list out that task. But deploying to heroku will raise the missing exception.

**Comment by ndbroadbent** [comment]
Please try updating to `0.3.14` and let me know if it's fixed

**Comment by drewB** [comment]
The error is is still there is 0.3.14

**Comment by drewB** [comment]
@runemadsen what version of the gem was working?

**Comment by runemadsen** [comment]
@drewB It happened when I bumped from `0.3.6` to `0.3.11`.

**Comment by drewB** [comment]
Interesting.  I still get the error on 0.3.6.

**Comment by pboling** [comment]
We've had the error on our deploys for at least a year, but not sure what versions.

**Comment by ndbroadbent** [comment]
Sorry I haven't been able to resolve this issue yet, would greatly appreciate a pull request if anyone can find a solution.

**Comment by mockdeep** [comment]
Just ran into this issue, too, when upgrading from `0.3.6` to `0.3.14`.

**Comment by mockdeep** [comment]
Just downgraded back to `0.3.6` to double check and I'm still seeing the error. I never noticed it before, but that's not to say it wasn't there. However, I did upgrade a bunch of gems today, so it may have been caused by that. Upgraded from rails `3.2.15` to `3.2.19`.

**Comment by charlesdeb** [comment]
Yep, same error here. Using rails 3.2.13, turbo-sprockets-rails3 (0.3.11). Even though I get the error, heroku still deploys a lot faster than without turbo-sprockets and it seems to deploy and launch fine.

**Comment by supermomonga** [comment]
Got same error when deploying heroku. below is the my env:
- Ruby `ruby 2.0.0p353 (2013-11-22 revision 43784) [x86_64-darwin13.2.0]`
- Rails `3.2.19`
- turbo-sprockets-rails3 `0.3.14`

`rake -T` on heroku:

```
❯ heroku run 'rake -T' | grep assets:clean 
Running `bundle exec rake -T` on myapp as run.2693:
rake assets:clean                  # Remove compiled assets
```

`rake -T` on local machine:

```
❯ bundle exec rake -T | grep assets:clean
rake assets:clean                  # Remove compiled assets
rake assets:clean_expired          # Remove old assets that aren't referenced by manifest.yml
```

**Comment by rrauenza** [comment]
I also see this issue when running the target from within supervisord:

<pre>
[program:clean_assets]
environment=RAILS_ENV="production"
command=/opt/rvm/bin/app_bundle exec rake assets:clean_expired                  
user=root                                                                       
autostart=true                                                                  
autorestart=false                                                               
startretries=0                                                                  
startsecs=0                                                                     
exitcodes=0                                                                     
directory=/opt/myapp                                              
stdout_logfile=/tmp/precompile_asserts.log                                      
redirect_stderr=true                
</pre>

**Comment by rrauenza** [comment]
Aha!

Add <code>require 'turbo-sprockets-rails3'</code> to config/application.rb per a vaguely similar http://stackoverflow.com/questions/7205964/dont-know-how-to-build-task-assetsprecompile

**Comment by forgotpw1** [comment]
I was able to resolve the heroku deploy error on rails 3.2.17 and turbo-sprockets-rails3 I had to set config.assets.handle_expired to true in the config/environments/production.rb file and add the config variable RAILS_GROUPS=assets to the heroku environment
